pkg/handler/v2/routes_v2.go

```diff
@@ -6900,3 +6922,138 @@ func (r Routing) deleteResourceQuota() http.Handler {
 		r.defaultServerOptions()...,
 	)
 }
+
+// TODO decide if swagger tag 'applications' is appropriate, or if it should also be added to 'projects'
+// swagger:route GET /api/v2/projects/{project_id}/clusters/{cluster_id}/applicationinstallations applications listApplicationInstallations
+//
+//     List ApplicationInstallations which belong to the given cluster
+//
+//     Produces:
+//     - application/json
+//
+//     Responses:
+//       default: errorResponse
+//       200: []ApplicationInstallation
+//       401: empty
+//       403: empty
+func (r Routing) listApplicationInstallations() http.Handler {
+	return httptransport.NewServer(
+		endpoint.Chain(
+			middleware.TokenVerifier(r.tokenVerifiers, r.userProvider),
+			middleware.UserSaver(r.userProvider),
+			middleware.SetClusterProvider(r.clusterProviderGetter, r.seedsGetter),
+		)(applicationinstallation.ListApplicationInstallations(r.userInfoGetter)),
+		applicationinstallation.DecodeListApplicationInstallations,
+		handler.EncodeJSON,
+		r.defaultServerOptions()...,
+	)
+}
+
+// swagger:route POST /api/v2/projects/{project_id}/clusters/{cluster_id}/applicationinstallations applications createApplicationInstallation
+//
+//     Creates ApplicationInstallation into the given cluster
+//
+//     Consumes:
+//     - application/json
+//
+//     Produces:
+//     - application/json
+//
+//     Responses:
+//       default: errorResponse
+//       201: ApplicationInstallation
+//       401: empty
+//       403: empty
+func (r Routing) createApplicationInstallation() http.Handler {
+	return httptransport.NewServer(
+		endpoint.Chain(
+			middleware.TokenVerifier(r.tokenVerifiers, r.userProvider),
+			middleware.UserSaver(r.userProvider),
+			middleware.SetClusterProvider(r.clusterProviderGetter, r.seedsGetter),
+		)(applicationinstallation.CreateApplicationInstallation(r.userInfoGetter)),
+		applicationinstallation.DecodeCreateApplicationInstallation,
+		handler.EncodeJSON,
```

Comment:
It is better to return 201 for any POST that creates resources in an async manner (also to be consistent with swagger above)
```suggestion
		handler.SetStatusCreatedHeader(handler.EncodeJSON),
```